### ISSUE TITLE
GC could not delete object that is split

### ISSUE THREAD

**Issue by carpawell** [nspcc-dev/neofs-node#1699]
Our mainnet node `asuka` reported such logs:

```
asuka neofs-node[71361]: 2022-08-17T16:10:04.625Z        warn        shard/gc.go:216        could not delete the objects        {"error": "object not found, split info has been provided"}
```

It seems like GC must not get such errors while removing objects.

1. Add object address to the error;
2. Do not [stop removing objects](https://github.com/nspcc-dev/neofs-node/blob/d99800ee93bbe9b0699313ba7a5ecab85699731c/pkg/local_object_storage/metabase/delete.go#L72) after fall (one of the shard kept many objects with GC and could do nothing with that);
3. Recheck that removing split objects works correctly.

NeoFS version: `v0.28.1`

**Comment by carpawell** [comment]
Reproduced on master. Split objects could not be deleted.

**Comment by fyrchik** [comment]
1. Consider indexing tomstone split id in metabase
2. Consider handling specific errors in GC

**Comment by carpawell** [comment]
Closed via #2019.
